From: "Torne (Richard Coles)" <torne@google.com>
Date: Mon, 25 Jul 2022 18:59:55 +0000
Subject: Suppress T SDK lints in Cronet.

Suppress lints for some uses of getIdentifier() in Cronet where there's
no alternative.

Bug: 1339376
Change-Id: I47be8650048300431e1da9e5b7f56e8722d237a2
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3779845
Auto-Submit: Richard Coles <torne@chromium.org>
Reviewed-by: Mohamed Heikal <mheikal@chromium.org>
Commit-Queue: Mohamed Heikal <mheikal@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1027878}
---
 .../cronet/android/api/src/org/chromium/net/CronetProvider.java | 2 ++
 .../src/org/chromium/net/smoke/CronetSmokeTestRule.java         | 1 +
 2 files changed, 3 insertions(+)

diff --git a/components/cronet/android/api/src/org/chromium/net/CronetProvider.java b/components/cronet/android/api/src/org/chromium/net/CronetProvider.java
--- a/components/cronet/android/api/src/org/chromium/net/CronetProvider.java
+++ b/components/cronet/android/api/src/org/chromium/net/CronetProvider.java
@@ -215,6 +215,8 @@ public abstract class CronetProvider {
      *         {@link #RES_KEY_CRONET_IMPL_CLASS} key.
      * @throws RuntimeException if the provider cannot be found or instantiated.
      */
+    // looking up resources from other apps requires the use of getIdentifier()
+    @SuppressWarnings("DiscouragedApi")
     private static boolean addCronetProviderFromResourceFile(
             Context context, Set<CronetProvider> providers) {
         int resId = context.getResources().getIdentifier(
diff --git a/components/cronet/android/test/smoketests/src/org/chromium/net/smoke/CronetSmokeTestRule.java b/components/cronet/android/test/smoketests/src/org/chromium/net/smoke/CronetSmokeTestRule.java
--- a/components/cronet/android/test/smoketests/src/org/chromium/net/smoke/CronetSmokeTestRule.java
+++ b/components/cronet/android/test/smoketests/src/org/chromium/net/smoke/CronetSmokeTestRule.java
@@ -107,6 +107,7 @@ public class CronetSmokeTestRule implements TestRule {
      *
      * @throws Exception if the class cannot be instantiated.
      */
+    @SuppressWarnings("DiscouragedApi")
     private void initTestSupport() throws Exception {
         Context ctx = InstrumentationRegistry.getTargetContext();
         String packageName = ctx.getPackageName();
-- 
2.37.3

